molecule/provisioners/libvirtprovisioner.py

```diff
@@ -0,0 +1,923 @@
+from pprint import pprint as pp
+import shlex
+import subprocess
+import tarfile
+import time
+
+try:
+    from lxml import etree as ET
+except ImportError:
+    try:
+        from xml.etree import cElementTree as ET
+    except ImportError:
+        from xml.etree import ElementTree as ET
+
+from molecule import utilities
+LOG = utilities.get_logger(__name__)
+
+try:
+    import guestfs
+    GUESTFS = True
+except ImportError:
+    GUESTFS = False
+    LOG.warning("\tPython module for libguestfs not available, certain networking-related functionality unavailable")
+
+from jinja2 import Template
+import libvirt
+import netaddr
+import requests
+
+from molecule.provisioners import baseprovisioner
+
+
+class LibvirtProvisioner(baseprovisioner.BaseProvisioner):
+    """
+    Implements a Molecule provisioner that uses libvirt to provision instances.
+    """
+
+    def __init__(self, molecule):
+        super(LibvirtProvisioner, self).__init__(molecule)
+        self._provider = self._get_provider()
+        self._platform = self._get_platform()
+        self._libvirt = libvirt.open(self.molecule.config.config['libvirt'][
+            'uri'])
+        self._pool_path = os.path.expanduser(
+            os.path.join('/opt/jenkins', 'libvirt', 'images'))
+        self._sources_path = os.path.expanduser(
+            os.path.join('/opt/jenkins', 'libvirt', 'sources'))
+        for path in [self._pool_path, self._sources_path]:
+            if not os.path.exists(path):
+                os.makedirs(path, 0775)
+        self._boot_wait = 60
+
+    def _get_provider(self):
+        return 'libvirt'
+
+    def _get_platform(self):
+        return self.default_platform
+
+    @property
+    def name(self):
+        return 'libvirt'
+
+    @property
+    def instances(self):
+        return self.molecule.config.config['libvirt']['instances']
+
+    @property
+    def default_provider(self):
+        return 'libvirt'
+
+    @property
+    def default_platform(self):
+        return ''
+
+    @property
+    def provider(self):
+        return self._provider
+
+    @property
+    def platform(self):
+        return self._platform
+
+    @property
+    def host_template(self):
+        return '{} ansible_ssh_host={} ansible_ssh_private_key_file={} ansible_ssh_user={}\n'
+
+    @property
+    def valid_providers(self):
+        return [{'name': 'Libvirt'}]
+
+    @property
+    def valid_platforms(self):
+        return [{'name': 'Libvirt'}]
+
+    @property
+    def ssh_config_file(self):
+        return '.molecule/ssh_config'
+
+    @property
+    def testinfra_args(self):
+        kwargs = {
+            'ansible-inventory':
+            self.molecule.config.config['ansible']['inventory_file'],
+            'connection': 'ansible'
+        }
+
+        return kwargs
+
+    @property
+    def serverspec_args(self):
+        return dict()
+
+    @property
+    def ansible_connection_params(self):
+        params = {'connection': 'ssh'}
+
+        return params
```

Comment:
You're doing this in a couple of places: you can remove lines by simply doing: "return {'connection': 'ssh'}".  It's a detail but reducing the overall line count is a mission that needs to be taken seriously :).